### DISCUSSION TITLE
Static assets list

### DISCUSSION THREAD

**Discussion by reabreu** [vercel/next.js#11102] [Help]
Hello all, I've been searching on the docs and couldn't find anything related to this so decided to open a question, so please consider the following scenario:

If I wanted to create a endpoint on my next.js app that exposes the assets the apps is using, on a per build basis, what would be the recommend way to do this? 

I noticed - correct me If I'm wrong - that all the assets used on client side are under the `.next/static` folder, and that there is a file `build-manifest.json` that list some of the js used on client side. However, since this is not considered a public facing API, I feel like going for an approach where we sniff this out might create problems in the future, if for however reason this structures changes.

Please advise if there is an official way to do this, and if not, is it something I can create a feature request for?

**Comment by timneutkens** [comment]
The main question I would have is why would you need this in particular?

**Reply by reabreu** [reply]
Hey, thanks for the fast reply. 

We have the scenario documented on a company blog post that you can read in detail here:
https://medium.com/ynap-tech/how-we-built-a-service-worker-for-our-micro-frontends-881eabfbf267

However the TLDR is:
Inside our k8s cluster, our domain specific apps (a few of them powered by next) must expose their assets list on a json format, which is consumed by our dynamic service worker generator service (named woz) to output a service worker based on the inbound request.
